browser/brave_shields/brave_shields_web_contents_observer.cc

```diff
@@ -166,7 +166,8 @@ void BraveShieldsWebContentsObserver::DispatchBlockedEventForWebContents(
     return;
   auto* shields_data_ctrlr =
       brave_shields::BraveShieldsDataController::FromWebContents(web_contents);
-  DCHECK(shields_data_ctrlr);
+  // sheilds_data-ctrlr can be null if the |web_contents| is generated in
```

Comment:
```suggestion
  // |shields_data_ctrlr| can be null if the |web_contents| is generated in
```

Reply:
I think this seems not related with component layer.
Standalone WebContents can be generated from anywhere. And we don't attach any tab-hepers to it.
I think this DCHECK is catched because WebContents used by playlist components created network request?

Reply:
> And we don't attach any tab-hepers to it.

Yeah, that looks what I wanted to say. |shields_data_ctrlr| is a tab helper and it's not attached when the contents is created from other layer than chrome layer. (tab_helpers.cc is in chrome layer). That's why it's null when request is fired from playlist's contents.